mirrord-auth/src/lib.rs

```diff
@@ -0,0 +1,100 @@
+#[cfg(feature = "webbrowser")]
+use std::time::Duration;
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+
+use lazy_static::lazy_static;
+#[cfg(feature = "webbrowser")]
+use rand::distributions::{Alphanumeric, DistString};
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+lazy_static! {
+    static ref FILE_DIR: PathBuf = [
+        std::env::var("HOME")
+            .or_else(|_| std::env::var("HOMEPATH"))
+            .unwrap_or_else(|_| "~".to_owned()),
+        ".mirrord_credentials".to_owned()
+    ]
+    .iter()
+    .collect();
+}
+
+#[derive(Deserialize, Serialize)]
+pub struct AuthConfig {
+    pub access_token: String,
+    pub refresh_token: Option<String>,
+}
+
+#[derive(Error, Debug)]
+pub enum AuthenticationError {
+    #[error(transparent)]
+    IoError(#[from] std::io::Error),
+    #[error(transparent)]
+    ConfigParseError(#[from] serde_json::Error),
+    #[error(transparent)]
+    #[cfg(feature = "webbrowser")]
+    ConfigRequestError(#[from] reqwest::Error),
+}
+
+type Result<T> = std::result::Result<T, AuthenticationError>;
+
+impl AuthConfig {
```

Comment:
Using of a raw file is quite "dangerous" - I know that many apps do it, but if it's possible to use a crate such as this https://github.com/hwchen/keyring-rs then let's do it (if not, open an issue to later transition, but would be easier to start with such solution)

Reply:
the 'username' for the keyring should be the OS username or something from cli input (with default value)?

Reply:
os username sounds good

Reply:
(in the example they use `whoami`)